code/game/gamemodes/antag_spawner.dm

```diff
@@ -127,6 +141,7 @@
 	var/TC_cost = 0
 	var/borg_to_spawn
 	var/list/possible_types = list("Assault", "Medical")
+	var/spawn_type = BORG_TELE

```

Comment:
Instead of making a magical define and switch on that, you could just move this code down from `/obj/item/weapon/antag_spawner/borg_tele` to, say, `/obj/item/weapon/antag_spawner/nukeops/borg_tele`. Then, in the places where you checked for `BORG_TELE`, you could call a proc (for example, `select_type()` or `spawn_antag()` and override its behavior on the children `/obj/item/weapon/antag_spawner/nukeops/borg_tele` and `/obj/item/weapon/antag_spawner/nukeops/ops_tele`.